mono/mini/interp/interp.c

```diff
@@ -5156,7 +5172,8 @@ interp_exec_method_full (InterpFrame *frame, ThreadContext *context, guint16 *st
 
 		   --sp;
 		   del = (MonoDelegate*)sp->data.p;
-		   sp->data.p = del->method_ptr;
+		   g_assert (del->interp_method);
+		   sp->data.p = del->interp_method;
```

Comment:
could we make the interp go through `del->method_ptr` as well too? it should be always set, right?

I don't like to introduce another field to `MonoDelegate`.

Reply:
It wouldn't work in mixed mode, the jit assumes method_ptr is a JIT entry point, while the intepreter needs it to be an InterpMethod.

Reply:
couldn't we go through the JIT entry point too with the interpreter?

After `CEE_MONO_LD_DELEGATE_METHOD_PTR` we emit `CEE_MONO_CALLI_EXTRA_ARG`, which is sort of special already. could we convert `CEE_MONO_CALLI_EXTRA_ARG` into a `MINT_JIT_CALL`?

Reply:
We could, but it would be very slow for the normal interpreter, since we would go interp->jit->interp.
MonoDelegate is not a very common object, it doesn't have to be very small.

Reply:
ok